tests/ResourceSyncingTest.php

```diff
@@ -126,6 +127,229 @@
     ], ResourceUser::first()->getAttributes());
 });
 
+// This tests attribute list on the central side, and default values on the tenant side
+// Those two don't depend on each other, we're just testing having each option on each side
+// using tests that combine the two, to avoid having an excessively long and complex test suite
+test('sync resource creation works when central model provides attributes and resource model provides default values', function () {
+    // when central model provides attributes => resoucre model will be created from the attribute values
```

Comment:
Is this trying to say that when the central model provides **a list of attributes**, the resource model will be created from those attributes' values?

Reply:
If so, then the comment is fine here, but it should be more clear by:
1. Using the same terminology everywhere. See the first lines of the comments I added in the last commit. `list of attributes`, `default values`, `mixed attribute list/defaults`, `no specified attributes`. If you can think of more accurate terms, we can use those, but these are the ones I used in the comments and think they're the most descriptive. The rest of the code (tests, implementation, comments, ...) should follow consistent terminology
2. Making it clear when the resource model "will be created". This is right above a a line that creates two models, so it sounds like it's being created here. Maybe it'd be more appropriate above the `attach()` call, to say that the resource user will be created based on attributes from `$centralModel`?

Reply:
>Making it clear when the resource model "will be created". This is right above a a line that creates two models, so it sounds like it's being created here.

FYI, I followed the rule to add comments when that specific part was going to start. 

> Maybe it'd be more appropriate above the attach() call, 

It makes sense right above the attach call. 
What about this second comment? L218. 
https://github.com/archtechx/tenancy/blob/93efa56f5932e93c6141af520cf4a78f67ff025e/tests/ResourceSyncingTest.php#L218-L222
Is it at the appropriate position? I'm also considering moving it to the right above the `ResourceUserProvidingAttributeNames::create` line.

Reply:
> FYI, I followed the rule to add comments when that specific part was going to start.

I don't understand what you mean by this.

> What about this second comment

1. I wouldn't use `=>`, I'd just make it a normal sentence. This may seem like a nitpick but it makes the comment more understandable at first look
2. I'd change `attributes` to `list of attributes` like I mentioned above — pick some terminology to follow everywhere. `list of attributes` is probably better than `attributes` because the latter could be understood as attribute names & values. Whereas `list of attributes` is clearly just a list of the attributes used.

> I'm also considering moving it to the right above the

Yes, move it there. I don't think `initialize()` is the appropriate place because I don't think `initialize()`  creates the models here.

Reply:
Update the comments and move them to appropriate positions.